benchmarks/perf-tool/add-filters-to-dataset.py

```diff
@@ -0,0 +1,172 @@
+import getopt
+import os
+import random
+import sys
+
+import h5py
+
+from osb.extensions.data_set import Context, HDF5DataSet
+
+"""
+Script builds complex dataset with additional attributes from exiting dataset that has only vectors. 
+Additional attributes are predefined in the script: color, taste, age. Only HDF5 format of vector dataset is supported.
+
+Script generates additional dataset of neighbours (ground truth) for each filter type. 
+
+Example of usage:
```

Comment:
What differentiates between filter and attribute?

Reply:
In this context I use attribute as an additional field for a document that we'll index, and filter is the set of criteria that will make a subset out of main set of documents. So this script can work in two modes: 
1. It takes existing set of vector data and adds fields of different types to each document. In this context I do use term attribute
2. Based on predefined rules it will take dataset generated at step 1 and apply filter to it, so the outcome is set of new datasets with true neighbors that are both ordered by similarity and also filtered. All those new datasets are stored in a separate new file.